### PR TITLE
add discount logic to PaymentPreviewView to render discounted price accordingly in template

### PR DESCRIPTION
**Pull request by Nuuggs**

### CODE REVIEW

File: student/tests/views/test_registration_views.py [hunk 1]
```diff
@@ -85,3 +88,45 @@ def test_registration_wizard_form_new_user(batch, existing_user):
     })
 
     assert User.objects.all().count() == 2
+
+def test_payment_preview_get_render_method_discount_logic():
```
Comment (samklq) [inline student/tests/views/test_registration_views.py:92]: might want to be a bit more specific like `test_payment_preview_get_passes_discounted_price_to_render_if_referral_code_valid`

File: student/tests/views/test_registration_views.py [hunk 2]
```diff
@@ -85,3 +88,45 @@ def test_registration_wizard_form_new_user(batch, existing_user):
...
+        batch=batch,
+        first_name='FirstName',
+        last_name='LastName',
+        email='user@email.com',
+        country_of_residence='SG',
+        referral_channel='word_of_mouth',
+        referral_code=coupon.code
+    )
+
+    registration = Registration.objects.get(pk=1)
```
Comment (samklq) [inline student/tests/views/test_registration_views.py:127]: pls put this above with the other 'arrange' statements! separate the action into it's own visual block.

File: student/views/registration.py [hunk 3]
```diff
@@ -71,8 +73,20 @@ def done(self, form_list, **kwargs):
...
             return redirect('basics_register')
 
+
 class PaymentPreviewView(View):
     def get(self, request, registration_id):
+        registration = Registration.objects.get(pk=registration_id)
+        payment_amount = settings.CODING_BASICS_REGISTRATION_FEE_SGD
+        if registration.referral_code:
+            coupon = Coupon.objects.get(code=registration.referral_code)
+            coupon_effect = coupon.effects.filter(couponable_type='Course', couponable_id=1).first()
```
Comment (samklq) [inline student/views/registration.py:83]: just a note to @rocketkai - we will change this to cater for list of discounts in future, probably when we handle tiered payments. will see how it goes.

File: student/views/registration.py [hunk 4]
```diff
@@ -83,7 +97,8 @@ def get(self, request, registration_id):
                 'payable_line_item_amount_in_dollars': settings.CODING_BASICS_REGISTRATION_FEE_SGD,
```
Comment (samklq) [inline student/views/registration.py:97]: might wanna take a look at what this variable was used for. if there are no more usages in the rest of the codebase after you've removed it, best practice is to clean this up too! can clarify over a call if needed.

File: student/views/registration.py (hunk 3 above)
Comment (rocketkai) [reply student/views/registration.py:83]: Noted, thank you!

**Pull request merged**